### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by abhijitdotsharma**
Login and Signup added
- Cart and Wishlist now redirect to Login Page if **not signed in**
- Logout Shown after user logged in successfully

Important files to check 
- `src/pages/login-page`
- `src/pages/signup-page`
- `src/context/auth-context.js`
- `src/components/RequiresAuth.js`

[View Deploy Preview](https://deploy-preview-7--nou-store-react.netlify.app/)

### CODE REVIEW

File: src/context/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+
+const AuthProvider = ({ children }) => {
+
+    //get userSignedInStatus from local storage to set it to useState
+    const userSignedInStatus = localStorage.getItem("userSignedInStatus")
+
+    //if userSignedInStatus is not yet set - returns null, state will take next value, i.e. false
+    const [isLoggedIn, setIsLoggedIn] = useState(userSignedInStatus || false);
```
Comment (harshalini) [inline src/context/auth-context.js:14]: Nice logic

**Review by harshalini — APPROVED** [review]
Code is readable and good, Go ahead to merge

File: src/pages/index.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import CartPage  from "./cart-page/CartPage";
 import LandingPage from "./landing-page/LandingPage";
```
Comment (sushil-mod) [inline src/pages/index.js:2]: I like how you export from a single file

**Review by sushil-mod — APPROVED** [review]
All looks good, but there are comments if it is not in use remove them and also console logs .

**Review by t007rushi — APPROVED** [review]
There is unwanted comments which are repetitive can remove some
Else Good to go ahead

**Pull request merged**